clippy_lints/src/doc/suspicious_doc_comments.rs

```diff
@@ -32,12 +33,14 @@ pub fn check(cx: &LateContext<'_>, attrs: &[Attribute]) -> bool {
     }
 }
 
-fn collect_doc_replacements(attrs: &[Attribute]) -> Vec<(Span, String)> {
+fn collect_doc_replacements(cx: &LateContext<'_>, attrs: &[Attribute]) -> Vec<(Span, String)> {
     attrs
         .iter()
         .filter_map(|attr| {
             if let Some((sym, com_kind)) = attr.doc_str_and_comment_kind()
                 && let AttrStyle::Outer = attr.style
+                // Filter out explicit `doc` attributes
+                && snippet_opt(cx, attr.span.source_callsite()).is_some_and(|s| s.starts_with('/'))
```

Comment:
If we do go with this, this could use the newer non-allocating snippet checking API
```suggestion
                && attr.span.source_callsite().check_source_text(cx, |s| s.starts_with('/'))
```

Reply:
Neat, I hadn't noticed this function, I've used it until we get the answer from @jdonszelmann.